src/OpenStudy/study_openinterface.jl

```diff
@@ -768,8 +822,34 @@ function configuration_parameter(
     return out
 end
 
-_cast(::Type{T}, val::T) where T = val
-function _cast(::Type{T}, val::String) where T
+"""
+    _cast(::Type{T}, val, default::T = _default_value(T))
+
+Converts `val` to type `T`, if possible.
+"""
+_cast(::Type{T}, val::T, default::T = _default_value(T)) where T = val
+_cast(::Type{String}, val::String, default::String = _default_value(String)) = val
+_cast(::Type{Int32}, val::Integer, default::Int32 = _default_value(Int32)) = Int32(val)
+_cast(::Type{Float64}, val::Float64, default::Float64 = _default_value(Float64)) = val
+_cast(::Type{Dates.Date}, val::Dates.Date, default::Dates.Date = _default_value(Dates.Date)) = val
+function _cast(::Type{T}, val::String, default::T = _default_value(T)) where T
     return parse(T, val)
 end
-_cast(::Type{Int32}, val::Integer) = Int32(val)
\ No newline at end of file
+_cast(::Type{Int32}, val::Integer, default::Int32 = _default_value(Int32)) = Int32(val)
+_cast(::Type{Dates.Date}, val::String, default::Dates.Date = _default_value(Dates.Date)) = _simple_date(val)
+
+_cast(::Type{T}, val::Nothing, default::T = _default_value(T)) where T = default
+
+"""
+    _cast_vector(::Type{T}, vector, default::T = _default_value(T))
+
+Converts `vector` to vector of type `T`, if possible.
+"""
+function _cast_vector(::Type{T}, vec::Vector{Any}, default::T = _default_value(T)) where T
+    out = T[]
+    for val in vec
+        push!(out, _cast(T, val, default))
+    end
+    return out
+end
```

Comment:
it should be
```julia
function _cast_vector(::Type{T}, vec::Vector{Any}, default::T = _default_value(T)) where T
    n = length(vec)
    out = Vector{T}(undef, n)
    for (i, val) in enumerate(vec)
        out[i] = _cast(T, val, default)
    end
    return out
end
```

Reply:
Feel free to update to that.
I just find it easier to write as is, and since this is not performance critical I dont really care...

Reply:
Same for all other comments related to this.
Feel free to switch...